Tests/Indicators/DerivativeOscillatorIndicatorTest.cs

```diff
@@ -0,0 +1,60 @@
+﻿using NUnit.Framework;
+using QuantConnect.Indicators;
+using System;
+using System.Linq;
+
+namespace QuantConnect.Tests.Indicators;
+
+public class DerivativeOscillatorIndicatorTest : CommonIndicatorTests<IndicatorDataPoint>
+{
+    protected override IndicatorBase<IndicatorDataPoint> CreateIndicator()
+    {
+        return new IndicatorDerivativeOscillator("Derivative Oscillator", 14, 5, 3, 9);
+    }
+
+    protected override string TestFileName => "spy_derivative_oscillator_daily_14_5_3_9.txt";
+
+    protected override string TestColumnName => "Derivative Oscillator 14 5 3 9";
+
+    [Test]
+    public void DoComputesCorrectly()
+    {
+        var derivativeOscillator = new IndicatorDerivativeOscillator(TestColumnName, 14, 5, 3, 9);
+
+        // List of random prices for testing
+        decimal[] prices = { 100m, 105m, 110m, 115m, 120m, 125m, 130m, 135m, 140m, 145m, 150m, 155m, 160m, 165m, 170m };
```

Comment:
Another good common practice in Lean is using `var` instead of explicitly typing variables when possible.